keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -664,6 +672,59 @@ endpoints:
 The `ServiceImport.Spec.IP` (VIP) can be used to access this service from within
 this cluster.
 
+
+#### Multi-network scenario
+One of more clusters in a ClusterSet can be running on a discrete network (a non-flat network). An MCS controller can use the `network.k8s.io` `ClusterProperty` to determine if a cluster in a `ClusterSet` is running on a discrete network. In this scenario, the generated EndpointSlice in an importing cluster will point to the FQDN (Fully Qualified Domain Name) of the Gateway/Loadbalancer of the cluster this `EndpointSlice` belongs to. The MCS controller implementation is responsible for determining the Gateway/Loadbalancer in a cluster for a non-flat network. A sample implementation can use a K8s [Gateway](https://gateway-api.sigs.k8s.io/) defined with an annotation `network.k8s.io: mcs-gateway` to match against a ServiceExport with the same annotation to determine the FQDN associated with it. This FQDN would then be used in the EndpointSlice generated for this cluster in the importing clusters. The MCS controller may implement creating the K8s Gateway and the necessary routing rules to automate the configuration.
```

Comment:
I think we need to relax this quite a bit. In multi-network, we do want to suggest that the `network.k8s.io` `ClusterProperty` is used, but that's probably where our spec ends.

We can use this as a place to reiterate that the endpoints in an `EndpointSlice` may only be representative of the pods backing the multi-cluster service and not real pod addresses, and may actually represent multiple pods (e.g. when a gateway is used).

Beyond that, we tend to favor flexibility. FQDNs addresses in `EndpointSlice`s cannot realistically be part of the spec because they don't actually do anything out-of-the-box. It might be better instead to have the implementation set the `ServceImport`'s `ClusterSetIP` to a gateway address, but that should probably be left to the implementation.

The main point we want to cover here is that so long as the rest of the spec is met, it doesn't matter how you wire up the backends.

Reply:
Great feedback thanks!